packages/sources/src/addSourceWizard/FinalWizard.js

```diff
@@ -17,39 +21,104 @@ const FinalWizard = ({
-                    <ErroredStep
-                        onRetry={ onRetry }
-                        onClose={ afterError }
-                        returnButtonTitle={ returnButtonTitle }
-                        message={errorMessage}
-                    />
-                    : <LoadingStep
-                        customText={<FormattedMessage id="wizard.loadingText" defaultMessage="Validating source credentials"/>}
-                        cancelTitle={<FormattedMessage id="wizard.cancel" defaultMessage="Cancel"/>}
-                    />,
-            isFinishedStep: true
-        }] }
-    />
-);
+    reset,
+    createdSource = {},
+    tryAgain
+}) =>{
+    const [ isDeletingSource, setIsDeleting ] = useState();
+    const [ isAfterDeletion, setDeleted ] = useState();
+
+    const intl = useIntl();
+    const { push } = useHistory();
+
+    const removeSource = () => {
+        setIsDeleting(true);
+
+        return getSourcesApi().removeSource(createdSource.id).then(() => setDeleted(true)).catch(() => setIsDeleting(false));
+    };
+
+    let step;
+    if (isAfterDeletion) {
+        step = <FinishedStep
+            onClose={ afterSubmit }
+            title={ intl.formatMessage({ id: 'wizard.removeSourceSuccessTitle', defaultMessage: 'Removing successful' }) }
+            successfulMessage={ intl.formatMessage({ id: 'wizard.removeSourceSuccessDescription', defaultMessage: 'Source was successfully removed.' }) }
+            hideSourcesButton={ hideSourcesButton }
+            returnButtonTitle={ returnButtonTitle }
+            secondaryActions={<Button variant="link" onClick={ reset }><FormattedMessage id="wizard.addAnotherSource" defaultMessage="Add another source"/></Button>}
+        />;
+    } else if (isDeletingSource) {
+        step = <LoadingStep
+            customText={<FormattedMessage id="wizard.removingSource" defaultMessage="Removing source"/>}
+        />;
+    } else if (isFinished) {
+        if (createdSource.applications[0]?.availability_status === 'available') {
+            step = <FinishedStep
+                onClose={ afterSubmit }
+                successfulMessage={ successfulMessage }
+                hideSourcesButton={ hideSourcesButton }
+                returnButtonTitle={ returnButtonTitle }
+                secondaryActions={<Button variant="link" onClick={ reset }><FormattedMessage id="wizard.addAnotherSource" defaultMessage="Add another source"/></Button>}
+            />;
+        } else if (createdSource.applications[0]?.availability_status === 'unavailable') {
+            step = <ErroredStep
+                onRetry={ onRetry }
+                onClose={ afterSubmit }
+                secondaryActions={<Button variant="link" onClick={ removeSource }>
+                    {intl.formatMessage({ id: 'wizard.removeSource', defaultMessage: 'Remove source' })}
+                </Button>}
+                primaryAction={() => push(`/sources/edit/${createdSource.id}`)}
```

Comment:
It's actually preferable to use link elements `a` for better accessibility. Is it possible to wrap this action into router link?

Reply:
added `component="a"`

Reply:
I don't think the `a` component will work it will use standard HTML `a` tag and probably will do full page refresh. You would have to prevent the event bubble (or use a router link).

Reply:
It works.

Reply:
Did you give it the `href` (to) property? That is the point of using the link element.

Reply:
Will fix it when the web works. :facepalm: